messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -88,6 +85,10 @@
   public static String clusterNamePrefix = "databusCluster";
   private Boolean readFromLocalStream;
   private int numList = 0;
+  private  List<String> collectorsWithReaders = new ArrayList<String>();
```

Comment:
The above case occurs in case if consumer read all the data from the existing collectors and waiting for new message(in next() method) and meanwhile if we added a new collector then consumer won't find new collector dynamically as createPartitionReaders() method won't be executed.

+1 for reusing the existing data structures.